### DISCUSSION TITLE
Is posible to use xxx.push with a pls file to play a entire playlist and then back to last...?

### DISCUSSION THREAD

**Discussion by caf2022** [savonet/liquidsoap#2796] [General]
Is posible to use xxx.push with a pls file to play a entire playlist and then back to last...?

I'm using this:

```
pedidos = request.queue(id="pedido")
pedido.push /home/music.mp3
```

And I did tests with a single mp3 file and works fine, is it possible to use a pls file with many mp3 files inside?

**Comment by toots** [comment]
Hi,

I don't think that this is currently possible but you can definitely implement your own telnet command for that. Something like this:
```
server.register(namespace=source.id(s), description="Queue a whole playlist on the source",
  "push_playlist", fun (pl) -> begin
    list.iter(fun (x) -> begin
      let (_, uri) = x
      s.push.uri(uri)
    end, playlist.parse(pl))
  end)
```

**Reply by caf2022** [reply]
Hi, thanks for the help. I tried the code but I'm having an error that I can't find what is.

```
server.register(namespace=source.id(s), description="Queue a whole playlist on the source",
Error 4: Undefined variable s
```
I have a default playlist that start with liquidsoap:

playlist = playlist(reload_mode="watch", loop=true, mode=random, "playlist.pls")

**Reply by smimram** [reply]
`s` is your source. You should replace it by `pedidos` I guess in your case.

**Reply by caf2022** [reply]
Get this error:

```
pedidos.push.uri(uri)
Error 5: this value has no method push
```
The code:

```
pedidos = request.queue(id="pedido")

server.register(namespace=source.id(pedidos), description="Queue a whole playlist on the source",
  "push_playlist", fun (pl) -> begin
    list.iter(fun (x) -> begin
      let (_, uri) = x
      pedidos.push.uri(uri)
    end, playlist.parse(pl))
  end)
```

**Reply by toots** [reply]
@caf2022 What version of liquidsoap are you running? This code is for the `2.x` version only.

**Reply by caf2022** [reply]
Hello, thanks for the help, my version is 2.0.7

**Reply by toots** [reply]
Ok `push.uri` does not exist for this version, it should be:
```
pedidos = request.queue(id="pedido")

server.register(namespace=source.id(pedidos), description="Queue a whole playlist on the source",
  "push_playlist", fun (pl) -> begin
    list.iter(fun (x) -> begin
      let (_, uri) = x
      pedidos.push(request.create(uri))
    end, playlist.parse(pl))
  end)
```
Doc for the operator at this version is here: https://www.liquidsoap.info/doc-2.0.7/reference.html#request.queue

**Reply by caf2022** [reply]
Hello, thanks fot the help, still with error:

```
pedidos.push(request.create(uri))
Error 5: this value has no method push
```
